src/org/mozilla/javascript/ScriptableObject.java

```diff
@@ -334,6 +334,38 @@ public void put(Symbol key, Scriptable start, Object value) {
         ensureSymbolScriptable(start).put(key, start, value);
     }
 
+    /**
+     * Set ( O, P, V, Throw = true)
+     *
+     * <p>https://262.ecma-international.org/12.0/#sec-set-o-p-v-throw
+     */
+    public static void putOrThrow(Context cx, Scriptable obj, String name, Object value) {
```

Comment:
These two Abstract Object Operations ought to be placed in AbstractEcmaObjectOperations. While doing so, check the guidelines at the top of the file

Reply:
ref. [204cd5f](https://github.com/mozilla/rhino/pull/1186/commits/204cd5fb8ba8f2e5a4c17bf7902fee722cf65260)
I made to move to `AbstractEcmaObjectOperations`.

Reply:
I rebased it.